commit/plugin_functions.go

```diff
@@ -628,6 +630,40 @@ func validateMerkleRootsState(
 	return true, nil
 }
 
+func backgroundReaderSync(
+	ctx context.Context,
+	wg *sync.WaitGroup,
+	lggr logger.Logger,
+	reader cciptypes.CCIPReader,
+	syncTimeout time.Duration,
+	ticker <-chan time.Time,
+) {
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				wg.Done()
+				return
+			case <-ticker:
+				timeoutCtx, cf := context.WithTimeout(ctx, syncTimeout)
+				updated, err := reader.Sync(timeoutCtx)
+				if err != nil {
+					lggr.Errorw("error syncing reader", "err", err)
+					cf()
```

Comment:
Why call `cf` if we want to continue?

Reply:
i cancel the timeout ctx that i initiate in that same loop. a new ctx with timeout is created in next iteration. without it we have resource leak

Reply:
ah understood, I misread it to be the cf for the `ctx` not the `timeoutCtx` 👍